src/Translator/assets/src/translator_controller.ts

```diff
@@ -7,168 +7,176 @@
+        TDomain extends DomainsOf<TMessage>,
+        TParameters extends ParametersOf<TMessage, TDomain>,
+    >(
+        id: TMessageId,
+        parameters: TParameters = {} as TParameters,
+        domain: RemoveIntlIcuSuffix<TDomain> | undefined = 'messages' as RemoveIntlIcuSuffix<TDomain>,
+        locale: LocaleOf<TMessage> | null = null
+    ): string {
+        if (typeof domain === 'undefined') {
+            domain = 'messages' as RemoveIntlIcuSuffix<TDomain>;
+        }
 
-    const translationsIntl = message.translations[`${domain}+intl-icu`];
-    if (typeof translationsIntl !== 'undefined') {
-        while (typeof translationsIntl[locale] === 'undefined') {
-            locale = localesFallbacks[locale] as LocaleOf<M>;
-            if (!locale) {
-                break;
-            }
+        if (typeof locale === 'undefined' || null === locale) {
+            locale = _locale as LocaleOf<TMessage>;
         }
 
-        if (locale) {
-            return formatIntl(translationsIntl[locale], parameters, locale);
+        const message = _messages[id] ?? (null as TMessage | null);
+        if (message === null) {
+            return id;
         }
-    }
 
-    const translations = message.translations[domain];
-    if (typeof translations !== 'undefined') {
-        while (typeof translations[locale] === 'undefined') {
-            locale = localesFallbacks[locale] as LocaleOf<M>;
-            if (!locale) {
-                break;
+        const translationsIntl = message.translations[`${domain}+intl-icu`] ?? undefined;
+        if (typeof translationsIntl !== 'undefined') {
+            while (typeof translationsIntl[locale] === 'undefined') {
+                locale = _localeFallbacks[locale] as LocaleOf<TMessage>;
+                if (!locale) {
+                    break;
+                }
+            }
+
+            if (locale) {
+                return formatIntl(translationsIntl[locale], parameters, locale);
+            }
+        }
+
+        const translations = message.translations[domain] ?? undefined;
+        if (typeof translations !== 'undefined') {
+            while (typeof translations[locale] === 'undefined') {
+                locale = _localeFallbacks[locale] as LocaleOf<TMessage>;
+                if (!locale) {
+                    break;
+                }
+            }
+
+            if (locale) {
+                return format(translations[locale], parameters, locale);
             }
         }
 
-        if (locale) {
-            return format(translations[locale], parameters, locale);
+        if (_throwWhenNotFound) {
+            throw new Error(`No translation message found with id "${id}".`);
         }
-    }
 
-    if (_throwWhenNotFound) {
-        throw new Error(`No translation message found with id "${message.id}".`);
+        return id;
     }
 
-    return message.id;
+    return {
+        setLocale,
+        getLocale,
+        setThrowWhenNotFound,
+        trans,
+    };
```

Comment:
Here I'm returning an object of methods instead of creating a `Translator` instance (implying this class exists), because:
- I want users being able to use `const { trans } = createTranslator()`, with an instance it does not work because `this` context is lost 
- It looks like a bit to the React Hooks or Vue Use systems, which has been pretty common for years in the JS ecosystem